src/android/Poynt.java

```diff
@@ -114,17 +230,232 @@ public void onActivityResult(int requestCode, int resultCode, final Intent data)
+                      byte[] byteArray = byteArrayOutputStream .toByteArray();
+                      String encoded = Base64.encodeToString(byteArray,Base64.DEFAULT); 
+                      cbk.success(encoded);  
+                    }
+                }
+
+            });
+        } catch (RemoteException e) {
+            e.printStackTrace();
+            this.callbackContext.error("Error");
+        }
+    }
+ 
+     
+    
+    public void showCollectAgreement(String msg) {
+        try {
+            final CallbackContext cbk=this.callbackContext;
+            Bundle options = new Bundle();
+            options.putString(Intents.EXTRA_LEFT_BUTTON_TITLE, "NO");
+            options.putString(Intents.EXTRA_RIGHT_BUTTON_TITLE, "ACCETTO");
+            /** AS URL **/
+            //options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_URL);
+            //String agreement = "https://www.visitamiapp.com/note-legali";
+            /* */
+           
+            //options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_HTML);
+            options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_URL);
+            String agreement = msg;
+//            /** AS TEXT **/
+//            options.putString(Intents.EXTRA_CONTENT_TYPE, Intents.EXTRA_CONTENT_TYPE_TEXT);
+//            String agreement = getAgreementText(R.raw.customer_agreement);
+            secondScreenService.captureAgreement(agreement,
+                    options, new IPoyntActionButtonListener.Stub() {
+                        @Override
+                        public void onLeftButtonClicked() throws RemoteException {
+                            cbk.success("NO");
+                        }
+
+                        @Override
+                        public void onRightButtonClicked() throws RemoteException {
+                            cbk.success("YES");
+                        }
+
+                    });
+        } catch (RemoteException e) {
+            e.printStackTrace();
+            this.callbackContext.error("ERROR");
+        }
+    }
+ 
+    /* ALE */
+    
+    /* ALE2 */
+    
+    
+     public String InfoStr(Business business)
+     {
+      return "{\"name\": \"" + business.getLegalName().replace("\"","") + "\"," +
+               "\"email\": \"" + business.getEmailAddress().replace("\"","") + "\"," +
+               "\"phone\": \"" + business.getPhone().getLocalPhoneNumber().replace("\"","") + "\","+
+               "\"mcc\": \"" + business.getMcc() +"\"}"; 
+     }
+    
+     public void showInfo() {
+        
+            final CallbackContext cbk=this.callbackContext;
+            IPoyntBusinessReadListener bizListener = new IPoyntBusinessReadListener.Stub() {
+            @Override
+            public void onResponse(Business business, PoyntError poyntError) throws RemoteException {
+                        if (business != null){
```

Comment:
As I mentioned in the email, please serialize the `business` object into JSON instead of returning a subset of it's properties.